File: go1.x/cookiecutter-aws-sam-eventbridge-hello-golang/{{cookiecutter.project_name}}/hello-world/model/aws/ec2/marshaller/marshaller.go
package marshaller

import (
    "encoding/json"
    "{{ cookiecutter.project_name }}/hello-world/model/aws/ec2"
)

func Marshal(inputEvent interface{}) ([]byte, error) {
    outputStream, err := json.Marshal(inputEvent)
    if err != nil {
        return nil, err
    }

    return outputStream, nil
}

func Unmarshal(inputStream []byte) (map[string]interface{}, error) {
    var outputEvent map[string]interface{}
    err := json.Unmarshal(inputStream, &outputEvent)
    if err != nil {
        return nil, err
    }

    return outputEvent, nil
}

func UnmarshalEvent(inputStream []byte) (ec2.AWSEvent, error) {
    var outputEvent ec2.AWSEvent
    err := json.Unmarshal(inputStream, &outputEvent)
    if err != nil {
        return ec2.AWSEvent{}, err
    }

    return outputEvent, nil
}

File: java8/event-bridge-gradle/{{cookiecutter.project_name}}/HelloWorldFunction/src/main/java/model/aws.ec2/marshaller/Marshaller.java
package model.aws.ec2.marshaller;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.type.TypeFactory;
import model.aws.ec2.AWSEvent;

public class Marshaller {

  private static final ObjectMapper MAPPER = createObjectMapper();

  public static <T> void marshal(OutputStream output, T value) throws IOException {
    MAPPER.writeValue(output, value);
  }

  public static <T> T unmarshal(InputStream input, Class<T> type) throws IOException {
    return MAPPER.readValue(input, type);
  }

  public static <T> AWSEvent<T> unmarshalEvent(InputStream input, Class<T> type) throws IOException {
    final TypeFactory typeFactory = MAPPER.getTypeFactory();
    return MAPPER.readValue(input, typeFactory.constructParametricType(AWSEvent.class, type));
  }

  private static ObjectMapper createObjectMapper() {
    return new ObjectMapper()
            .findAndRegisterModules()
            .enable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
  }
}
